rust/arrow/src/array/array.rs

```diff
@@ -1078,11 +1240,31 @@ impl From<Vec<&[u8]>> for BinaryArray {
     }
 }
 
-impl<'a> TryFrom<Vec<Option<&'a str>>> for BinaryArray {
+impl From<Vec<&[u8]>> for StringArray {
+    fn from(v: Vec<&[u8]>) -> Self {
+        let mut offsets = Vec::with_capacity(v.len() + 1);
+        let mut values = Vec::new();
+        let mut length_so_far = 0;
+        offsets.push(length_so_far);
+        for s in &v {
+            length_so_far += s.len() as i32;
```

Comment:
Should we add check here? The `&[u8]` maybe illegal utf8 string

Reply:
I'm not sure of how to check, but I've come across https://www.reddit.com/r/rust/comments/8jvk37/validating_utf8_strings_using_as_little_as_07/

I'll take a look into checking, I agree that we should do the check

Reply:
Having thought of this, I'm in favour of removing this. My rationale's that the responsibility to ensure that data is valid should rest on the user, otherwise we'd be validating already valid UTF8 data, and potentially sacrificing some performance.

`BinaryArray::from(&[u8])` would remain. Any thoughts @liurenjie1024 @paddyhoran?
If the user knows that their data is valid UTF8, they can cast the data by `StringArray::from(binary_array.data())`.

Reply:
Sounds reasonable to me.

Reply:
Yep, I agree.